### ISSUE TITLE
Magnet links problem results in "Couldn't add torrent: invalid or corrupt torrent file"

### ISSUE THREAD

**Issue by diegoviola** [fagga/transmission-remote-cli#151]
When trying to add a magnet link such as this one:
```
magnet:?xt=urn:btih:b425907e5755031bda4a8d1b6dccaca97da14c04&dn=archlinux-2015.01.01-dual.iso&tr=udp://tracker.archlinux.org:6969&tr=http://tracker.archlinux.org:6969/announce
```
I get the following in transmission-remote-cli:

```
Couldn't add torrent "magnet:?xt=urn:btih:b425907e5755031bda4a8d1b6dcc
aca97da14c04&dn=archlinux-2015.01.01-dual.iso&tr=udp://tracker.archlin
ux.org:6969&tr=http://tracker.archlinux.org:6969/announce":           
invalid or corrupt torrent file                                       
```
I add the magnet link by pressing A (capital a) on transmission-remote-cli, then I paste that URI, and I get the error above.

However, this works:

```
transmission-remote-cli "magnet:?xt=urn:btih:b425907e5755031bda4a8d1b6dccaca97da14c04&dn=archlinux-2015.01.01-dual.iso&tr=udp://tracker.archlinux.org:6969&tr=http://tracker.archlinux.org:6969/announce"
```
Pressing A (capital a) and just using the hash ```btih:b425907e5755031bda4a8d1b6dccaca97da14c04``` also works.

If I try adding a shortened magnet such as: ```magnet:?xt=urn:btih:b425907e5755031bda4a8d1b6dccaca97da14c04``` it doesn't work also.

**Comment by quillkiragimmy** [comment]
Magnet links are used like torrent files. Just press 'a' then paste the link. :)

**Comment by diegoviola** [comment]
That works, but why is A (capital a) not working?

**Comment by quillkiragimmy** [comment]
Maybe the developer thought magnet link should be a 'link', not hash. The program just add the prefix (magnet:?xt=urn:btih:) to the hash and then send it to the server in exactly the same way as URL & filePath.

**Comment by fagga** [comment]
Yes, a hash is not a link. But the current way to handle this with two
different shortcuts is silly. I removed the capital A shortcut and it should
now autodetect if something is a naked hash or not.

**Comment by diegoviola** [comment]
Thanks.

**Comment by quillkiragimmy** [comment]
You forgot to update the help menu entry :).

**Comment by fagga** [comment]
Puddlepoop! May the heavens curse my crappy design!

**Comment by lampslave** [comment]
However, using of common shortcut is cause for autocomplete issues. For example, if I want paste magnet or hash, I need to remove `~/` first. And if magnet or hash starts with `\t`, `~/` appears again (adding [here](https://github.com/fagga/transmission-remote-cli/blob/master/transmission-remote-cli#L3049) `and index > 0` can solve this).

**Comment by mfonville** [comment]
I get the same error using 'transgui' (Transmission Remote GUI 5.0.1) when clicking magnet links. My feeling is that this error is thus not caused per se by the remote client, but by the transmission daemon?
